src/include/duckdb/storage/storage_extension.hpp

```diff
@@ -26,11 +27,18 @@ struct StorageExtensionInfo {
 typedef unique_ptr<Catalog> (*attach_function_t)(AttachedDatabase &db, const string &name, AttachInfo &info,
                                                  AccessMode access_mode);
 typedef unique_ptr<TransactionManager> (*create_transaction_manager_t)(AttachedDatabase &db, Catalog &catalog);
+typedef unique_ptr<TableFunctionRef> (*create_database_t)(ClientContext &context, const string &extension_name,
```

Comment:
We probably don't need to pass the `extension_name` anymore as its primary use case would be to look up the extension name from the list of registered storage_extensions

Reply:
Done